.buildkite/dev-docker/manifest.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+source .buildkite/retry.sh
+
+set +x
+
+BUILD_FROM_BRANCH=$(buildkite-agent meta-data get BUILD_FROM_BRANCH)
+if [[ -z "${BUILD_FROM_BRANCH}" ]]; then
+    BUILD_FROM_BRANCH=${BUILDKITE_BRANCH}
+fi
+PUSH_LATEST=$(buildkite-agent meta-data get PUSH_LATEST)
+PUBLIC_DOCKER_REPO=$(buildkite-agent meta-data get PUBLIC_DOCKER_REPO)
+
+# login to docker registry
+DOCKER_PASSWORD=$(vault read -field token /secret/ci/elastic-rally/release/docker-hub-rally)
+retry 5 docker login -u elasticmachine -p $DOCKER_PASSWORD
+unset DOCKER_PASSWORD
```

Comment:
We're missing Elastic Docker registry variant here. Would be good to avoid repetition with `run.sh` too, maybe a shared file, like `retry.sh` ?

Reply:
Good idea, I've merged both `run.sh` and `manifest.sh` into a single script